language-design.md

```diff
@@ -17,7 +17,41 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
-
+Quote 1: "Why shouldn’t those 'interfaces' be as humanely designed as the ones we tap and swipe?" [Pavlus, 2012].
+
+Here Pavlus asks why programming languages' interfaces are not as "humanely
+designed" as software's graphical interfaces are. This quote stood out to us
+because Pavlus is trying to make a point, but actually ends up asking a
+question that is very easy to answer! One answer is that in general, software
+(programming languages included) is designed towards its users. The software
+Pavlus is referring to is probably targeted at a much broader audience, so it
+needs to be easier. Also, we take issue with Pavlus implying that programming
+is comparable to tapping and swiping. Is Pavlus claiming that programming
+should be as easy as Angry Birds? I doubt that would be a productive language.
+
+Quote 2: "Public APIs, like diamonds, are forever" [Bloch, 2006].
+
+This is not a feeling that we are used to. I (Ben) have never written anything
+like this, anything that would be public-facing and would cause an outcry if
+changed. It is interesting to think how choices made at the outset could affect
+your language fifteen years later, possibly. In fact, every future time you
+change something, you'd have to ensure backwards compatibility. It really
+emphasizes how thoughtful one needs to be in designing a language or an API. It
+is in general much better to wait until you are ready than to send out a rushed
+product.
+
+Quote 3: "If the design of the Java programming language as it is now had been put forth three years ago, it would have failed — of that I am sure" [Steele, 1998].
+
+This quote is interesting because although this was written 17 years ago, 
+we can see this happening with languages being chosen for teaching introductory 
+programming. For most of the late 2005-2010, Java was the main language 
+that was used to teach programming. This can be seen from College Board 
+choosing to move from C++ to Java for the Computer Science AP class. Now 
+that Java has had a lot added to it over the years, it is the old, hard to 
+enter language. Smaller languages like Python and Ruby have started to take 
+the place as introductory languages mostly because of their simplicity. This 
+should make us think about how additions, while great for existing users of 
+the DSL, might harm acquiring new users. 
 
```

Comment:
Cool, do you think that python has the same potential for that as Java/C++ did? Interesting to note that Java/C++ are both heavily influenced by C